src/Sylius/Behat/Context/Ui/Admin/ManagingZoneContext.php

```diff
@@ -0,0 +1,422 @@
+     */
+    public function iDoNotAddACountryMember()
+    {
+        // Intentionally left blank to fulfill context expectation
+    }
+
+    /**
+     * @When I add a country :name
+     * @When I add a province :name
+     * @When I add a zone :name
+     */
+    public function iAddACountry($name)
+    {
+        $this->createPage->addMember();
+        $this->createPage->chooseMember($name);
+    }
+
+    /**
+     * @When I add it
+     * @When I try to add it
+     */
+    public function iAddIt()
+    {
+        $this->createPage->create();
+    }
+
+    /**
+     * @When I save my changes
+     */
+    public function iSaveMyChanges()
+    {
+        $this->updatePage->saveChanges();
+    }
+
+    /**
+     * @Then I should be notified about successful creation
+     */
+    public function iShouldBeNotifiedAboutSuccessfulCreation()
+    {
+        $this->notificationChecker->checkCreationNotification(self::RESOURCE_NAME);
+    }
+
+    /**
+     * @Then I should be notified about successful edition
+     */
+    public function iShouldBeNotifiedAboutSuccessfulEdition()
+    {
+        $this->notificationChecker->checkEditionNotification(self::RESOURCE_NAME);
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully deleted
+     */
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyDeleted()
+    {
+        $this->notificationChecker->checkDeletionNotification(self::RESOURCE_NAME);
+    }
+
+    /**
+     * @Then /^the (zone named "[^"]+") with (the "[^"]+" country member) should appear in the registry$/
+     * @Then /^the (zone named "[^"]+") with (the "[^"]+" province member) should appear in the registry$/
+     * @Then /^the (zone named "[^"]+") with (the "[^"]+" zone member) should appear in the registry$/
+     */
+    public function theZoneWithTheCountryShouldAppearInTheRegistry(ZoneInterface $zone, ZoneMemberInterface $zoneMember)
+    {
+        $this->assertZoneAndItsMember($zone, $zoneMember);
+    }
+
+    /**
+     * @Then /^(this zone) should have only (the "([^"]*)" country member)$/
+     * @Then /^(this zone) should have only (the "([^"]*)" province member)$/
+     * @Then /^(this zone) should have only (the "([^"]*)" zone member)$/
+     */
+    public function thisZoneShouldHaveOnlyTheProvinceMember(ZoneInterface $zone, ZoneMemberInterface $zoneMember)
+    {
+        $this->assertZoneAndItsMember($zone, $zoneMember);
+
+        Assert::eq(
+            1,
+            $this->updatePage->countMembers(),
+            sprintf('Zone %s should have only %s zone member',$zone->getName(), $zoneMember->getCode())
+        );
+    }
+
+    /**
+     * @Then /^(this zone) name should be "([^"]*)"/
+     */
+    public function thisZoneNameShouldBe(ZoneInterface $zone, $name)
+    {
+        Assert::true(
+            $this->updatePage->hasResourceValues(['code' => $zone->getCode(), 'name' => $name]),
+            'Zone name should be modified but it is not'
```

Comment:
Maybe more meaningful will be `sprintf('Zone should be named %s, but is %s', $name, $zone->getName())`